src/rulesets/Base/arraymath.jl

```diff
@@ -19,28 +19,49 @@ end
 ##### `*`
 #####
 
+
 function rrule(
     ::typeof(*),
     A::AbstractVecOrMat{<:CommutativeMulNumber},
     B::AbstractVecOrMat{<:CommutativeMulNumber},
 )
     project_A = ProjectTo(A)
     project_B = ProjectTo(B)
+    function times_pullback(ȳ)
+        Ȳ = unthunk(ȳ)
+        dA = @thunk(project_A(Ȳ * B'))
```

Comment:
Is it work doing something like this here? 

```
maybe_mul!(A, B) = A * B
maybe_mul!(A::StridedArray, B::StridedArray) = (X = similar(B); mul!(X, A, B)) 
```

Or just push that discussion to https://github.com/JuliaDiff/ChainRulesCore.jl/issues/411

Reply:
> Or just push that discussion to JuliaDiff/ChainRulesCore.jl#411

This